### PR TITLE
[SQSERVICES-1773] Allow pagination for team search endpoint

### PR DESCRIPTION
**Pull request by battermann**
https://wearezeta.atlassian.net/browse/SQSERVICES-1773

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

**Review by mdimjasevic — COMMENTED** [review]
Do you want to update JSON golden test files with an example where `searchPagingState` is a `Just` value? It's a bit weird to put `Nothing`s in 20 existing cases and to none with a `Just` value.

**Comment by battermann** [comment]
> Do you want to update JSON golden test files with an example where `searchPagingState` is a `Just` value? It's a bit weird to put `Nothing`s in 20 existing cases and to none with a `Just` value.

You are right. Done!

**Review by mdimjasevic — COMMENTED** [review]
Are you aware of the `Wire.API.Routes.MultiTablePaging` module and the paging facilities provided by it? If yes, why rolling your own pagination system?

File: services/brig/src/Brig/User/API/Search.hs [hunk 1]
```diff
@@ -136,7 +137,7 @@ searchLocally searcherId searchTerm maybeMaxResults = do
   esResult <-
     if esMaxResults > 0
       then Q.searchIndex (Q.LocalSearch searcherId searcherTeamId teamSearchInfo) searchTerm esMaxResults
-      else pure $ SearchResult 0 0 0 [] FullSearch
+      else pure $ SearchResult 0 0 0 [] FullSearch Nothing
```
[the remark this answers is not in the document]
Comment (battermann) [reply services/brig/src/Brig/User/API/Search.hs:140]: If it is `Nothing` then the behavior doesn't change. This function is part of the `ConnectionAPI` and this API is not subject to change in this ticket/PR. The only reason to pass as parameter is to make it more visible.

File: services/brig/src/Brig/User/API/Search.hs [hunk 2]
```diff
@@ -113,7 +113,8 @@ searchRemotely domain searchTerm = do
         searchFound = count,
         searchReturned = count,
         searchTook = 0,
-        searchPolicy = S.searchPolicy searchResponse
+        searchPolicy = S.searchPolicy searchResponse,
+        searchPagingState = Nothing
```
[the remark this answers is not in the document]
Comment (battermann) [reply services/brig/src/Brig/User/API/Search.hs:117]: If it is `Nothing` then the behavior doesn't change. This function is part of the `ConnectionAPI` and this API is not subject to change in this ticket/PR. The only reason to pass as parameter is to make it more visible.

**Review by mdimjasevic — APPROVED** [review]
Thanks for clarifying! I'd say this is good to go.

**Pull request merged**